raiden_libs/test/mocks/client.py

```diff
@@ -0,0 +1,236 @@
+from eth_utils import is_checksum_address, remove_0x_prefix, is_same_address, encode_hex
+from coincurve import PrivateKey
+from functools import wraps
+from sha3 import keccak_256
+from web3.utils.events import get_event_data
+from web3 import Web3
+import logging
+
+from raiden_contracts.contract_manager import get_event_from_abi
+
+from raiden_libs.utils import private_key_to_address, make_filter
+from raiden_libs.messages import BalanceProof
+
+
+log = logging.getLogger(__name__)
+NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+
+def get_event_logs(
+        web3: Web3,
+        contract_abi: dict,
+        event_name: str,
+        fromBlock=0,
+        toBlock=None
+):
+    """Helper function to get all event logs in a given range"""
+    abi = get_event_from_abi(contract_abi, event_name)
+    tmp_filter = make_filter(web3, abi, fromBlock=0, toBlock=toBlock)
+    entries = tmp_filter.get_all_entries()
+    web3.eth.uninstallFilter(tmp_filter.filter_id)
+    return entries
+
+
+def sha3(data):
+    return keccak_256(data).digest()
+
+
+def assert_channel_existence(func):
+    """Trigger an assert if there is no channel registered with an other participant"""
+    @wraps(func)
+    def func_wrapper(self, partner_address, *args, **kwargs):
+        assert is_checksum_address(partner_address)
+        assert partner_address in self.partner_to_channel_id
+        return func(self, partner_address, *args, **kwargs)
+    return func_wrapper
+
+
+def sync_channels(func):
+    """Synchronize all channels opened with a client"""
+    @wraps(func)
+    def func_wrapper(self, *args, **kwargs):
+        self.partner_to_channel_id = self.sync_open_channels()
+        return func(self, *args, **kwargs)
+    return func_wrapper
+
+
+class MockRaidenNode:
+    def __init__(self, privkey, channel_contract):
+        self.privkey = privkey
+        self.address = private_key_to_address(privkey)
+        self.contract = channel_contract
+        self.partner_to_channel_id = dict()
+        self.token_network_abi = None
+        self.token_contract = None
+        self.client_registry = dict()
+        self.web3 = self.contract.web3
+
+    @sync_channels
+    def open_channel(self, partner_address) -> int:
+        """Opens channel with a signle partner
+        Parameters:
+            partner_address (str) - a valid ethereum address of the partner
```

Comment:
type annotation instead of type in docs

Reply:
(I'll eventually change address type to `NewType` like what you have in PFS)

Reply:
We should add these basic types to the libs as well.